### ISSUE TITLE
Multi-scrobbler only submits the last 20 cached scrobbles to Maloja using Last.fm as a source

### ISSUE THREAD

**Issue by maluvola** [FoxxMD/multi-scrobbler#153]
- multi-scrobbler: [0.6.5, no docker or flatpak]
- maloja [3.2.2]

**Comment by FoxxMD** [comment]
Hello

This is by design. multi-scrobbler is not an application for importing your historical scrobbles from LastFM. If you'd like to import all of your scrobbles from LastFM please follow the directions in the [Maloja documentation](https://github.com/krateng/maloja?tab=readme-ov-file#data).

multi-scrobbler is designed to be run continuously so it can monitor scrobbles in real time. The 20 scrobble value is in place to help "fill in the gaps" in the event it crashes or is stopped for some reason -- the cache is used to get any missed scrobbles during that down time. If this use case, getting that last 20 scrobbles, is not adequate for you please describe where it is failing and I may be able to address it.
 
Additionally, since you are running MS natively, you can adjust this value yourself to get more "cached" scrobbles, if you would like:

* stop multi-scrobbler, if it is running
* replace [`limit` in `LastfmSource.ts`](https://github.com/FoxxMD/multi-scrobbler/blob/master/src/backend/sources/LastfmSource.ts#L76) and replace [`this.MAX_STORED_SCROBBLES` on this line in `MalojaScrobbler.ts`](https://github.com/FoxxMD/multi-scrobbler/blob/master/src/backend/scrobblers/MalojaScrobbler.ts#L333) with the number of cached scrobbles you want
* restart multi-scrobbler by running `npm run start`

**Comment by maluvola** [comment]
Thank you. The reason I want to increase the number of cached scrobbles is because I don't let the server on at all times, and I like it mirrored with Last.fm, so I can let them be submitted to Maloja after listening to music on my phone.

**Comment by maluvola** [comment]
Thank you. The reason I want to increase the number of cached scrobbles is that I don't keep the server on at all times, and I like to have it mirrored with Last.fm. This way, scrobbles can be submitted to Maloja after I listen to music on my phone.

**Comment by FoxxMD** [comment]
Then increasing the limits I mentioned should accomplish your goal. Be aware that Last.FM will return a [maximum of 200 scrobbles](https://www.last.fm/api/show/user.getRecentTracks) so that's as high as the limit in `LastfmSource.ts` can go. You should make sure to start multi-scrobbler before you have scrobbled that many tracks since last using it.

**Comment by FoxxMD** [comment]
@maluvola please try pulling the `develop` branch. It has these changes in it:

* sources pull the maximum # of listens on startup (200 for last.fm)
* clients pull the maximum # of scrobbles for comparison on startup (100 for maloja)
* source and client configs can specify these numbers for startup if the default maximum is not desirable:

ex lastfm.json
```json
[
  {
    "name": "mylfm",
    "enable": true,
    "configureAs": "source",
    "data": {
    },
    "options": {
      "scrobbleBacklogCount": 100
    }
  }
]
```
ex maloja.json

```json
[
  {
    "name": "default",
    "enable": true,
    "data": {
    },
    "options": {
      "refreshInitialCount": 20,
    }
  }
]
```

I believe this should address your issue?

**Comment by maluvola** [comment]
Yes, it should. Does your previous answer work as well? I made the changes but haven't listened to more than a dozen tracks yet.

**Comment by FoxxMD** [comment]
Yes my previous answer will still work. The new solution in `develop` is a configuration-based version of my previous answer. I'd suggest switching to that, eventually, so that you can continue to pull updates to MS without having to resolve file changes.

**Comment by FoxxMD** [comment]
Released in [v0.8.0](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.0)
